pkg/apis/certificates/types.go

```diff
@@ -43,6 +43,10 @@ type CertificateSigningRequestSpec struct {
 	// Base64-encoded PKCS#10 CSR data
 	Request []byte `json:"request"`
 
+	// signingProfile specifies a requested signing profile for a CA to use
+	// when issuing the certificate.
+	// SigningProfile string `json:"signingProfile,omitempty"`
```

Comment:
I'd rather let the requester specify the things they want (definitely usages, maybe expiry), rather than making them reference a signing profile.
* not sure we want to let all clients view signing profiles
* a client should be able to request a usage there isn't a profile for yet, and an approver create the signing profile in response
* a client shouldn't have to try to find the tightest bounding match between existing signing profiles and the permissions they need